### PR TITLE
Add indent option - fixes #23

### LINKED ISSUE #23: Add `indent` option
It would be nice to be able to control what is set to cursorTo in order to support indentation of rows, if desired.

### PR DESCRIPTION
**Pull request by diogofsilva**
Fixes #23

### CODE REVIEW

**Comment by satazor** [comment]
Sorry I've commented on your committed directly instead on the PR, will fix it now.

File: test.js [hunk 1]
```diff
@@ -121,6 +121,20 @@ test('.stopAndPersist() - with new symbol and text', macro, spinner => {
 	spinner.stopAndPersist({symbol: '@', text: 'all done'});
 }, /@ all done/);
 
+test('spinner with default indent (0)', t => {
+	const spinner = new Ora('foo');
+	t.falsy(spinner.indent);
+});
+
+test('spinner with indent (7)', t => {
```
Comment (satazor) [inline test.js]: I think we should test if the actual output is indented to 7 spaces. Additionally I would remove the test above (0).. I feel it is unnecessary.

File: readme.md [hunk 2]
```diff
@@ -68,6 +68,13 @@ Values: `black` `red` `green` `yellow` `blue` `magenta` `cyan` `white` `gray`
 
 Color of the spinner.
 
+##### indent
+
+Type: `number`<br>
+Default: `0`<br>
```
Comment (sindresorhus) [inline readme.md]: The `<br>` is not needed on this line.

File: test.js (hunk 1 above)
Comment (diogofsilva) [inline test.js]: I'm struggling a bit with testing the actual indented output. I was trying something along the lines of the code below, but with no luck.

```javascript
test('indented spinner', async t => {
    const stream = getPassThroughStream();
    const output = getStream(stream);

    const spinner = new Ora('foo');
    spinner.stream = stream;
    spinner.indent = 3;
    spinner.color = false;
    spinner.enabled = true;
    spinner.start();

    stream.end();

    t.regex(await output, /\s{3}${spinnerChar} foo/);
});
```

This is the ava output:

```
   145:                                                    
   146:   t.regex(await output, /\s{3}${spinnerChar} foo/);
   147: });                                                

  Value must match expression:

    "⠋ foo"

  Regular expression:

    /\s{3}${spinnerChar} foo/

  awaitoutput
  => "⠋ foo"

  Test.<anonymous> (test.js:146:4)
  step (test.js:26:191)
```

Can someone give me a hand or point me in the right direction?

Comment (sindresorhus) [inline test.js]: You need to understand what your changes in this PR do. `this.stream.cursorTo(this.indent);` doesn't indent the string with spaces, but instead moves the CLI cursor. So you need to check for the correct ansi escape codes and that they match the indent option value.

**Comment by sindresorhus** [comment]
ping @diogofsilva

**Comment by sindresorhus** [comment]
Closing for lack of response.

**Pull request closed without merging**